### DISCUSSION TITLE
Sensor channel and attribute doesn't fit

### DISCUSSION THREAD

**Discussion by mejtejt** [zephyrproject-rtos/zephyr#29630] [Q&A] [answered]
Hi, 
If I want to use external RTC which operates through I2C, would you suggest using sensor driver api's like `sensor_driver_api`?
In that case I can't use neither one of the sensor_channel enums for example in function `m4lt62_channel_get`, but I have to read seconds, minutes, hours etc.. 
Do you have some other approaches in mind ?

**Comment by henrikbrixandersen** [comment, marked as answer]
You can add sensor-specific channels, attributes, and triggers. See e.g. https://github.com/zephyrproject-rtos/zephyr/blob/master/include/drivers/sensor/mcux_acmp.h

**Comment by henrikbrixandersen** [comment]
Another solution could be to use the counter API similar to how the DS3231 RTC driver is implemented: https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/counter/Kconfig.maxim_ds3231
